src/Maestro/SubscriptionActorService/PullRequestActor.cs

```diff
@@ -742,10 +742,10 @@ private async Task<string> CreatePullRequestAsync(List<UpdateAssetsParameters> u
             (string targetRepository, string targetBranch) = await GetTargetAsync();
             IRemote darcRemote = await DarcRemoteFactory.GetRemoteAsync(targetRepository, Logger);
 
-            List<(UpdateAssetsParameters update, List<DependencyUpdate> deps)> requiredUpdates =
+            TargetRepoDependencyUpdate repoDependencyUpdate =
                 await GetRequiredUpdates(updates, DarcRemoteFactory, targetRepository, targetBranch);
 
-            if (requiredUpdates.Count < 1)
+            if (repoDependencyUpdate.CoherencyCheckSuccessful && repoDependencyUpdate.RequiredUpdates.IsNullOrEmpty())
```

Comment:
This looks like we'd create a PR if the coherency check currently fails? If so, this is great. Also, do we have a test for this corner case?

Reply:
>This looks like we'd create a PR if the coherency check currently fails? If so, this is great.

Yes. From [the comment](https://github.com/dotnet/arcade-services/pull/2266#issuecomment-1515177854): "We allow coherency to fail and still do an update PR. This is not the case today."

> Also, do we have a test for this corner case?

I have added a TestScenario `ScenarioTests_GitHubFlow.Darc_GitHubFlow_NonBatched_FailingStrictCoherency` that checks that the PR is opened while the coherency algorithm failed. And I've modified(+renamed) the existed test `PullRequestActorTests.UpdateWithAssetsWhenStrictAlgorithmFails`

Reply:
The specific case here that I'm interested in is when there are no updates (except potential coherencyupdates), and coherency fails.